conda_smithy/templates/run_docker_build.tmpl

```diff
@@ -26,9 +26,12 @@ show_channel_urls: true
 CONDARC
 )
 
+HOST_USER_ID=$( hash docker-machine 2>/dev/null && docker-machine ssh $(docker-machine active) 'id -u $(whoami)' || id -u $(whoami) )
```

Comment:
To be absolutely clear, I *was* talking about the user inside the container. In many respects I'm very comfortable if it is necessary to run the container as root, but within the container we should be able to build (most) recipes as a non root user.

I have another question that will highlight my ignorance, but I'd like to know whether starting the container as root (as we currently do) and simply ``su`` to a different user has been considered. Conceptually it feels considerably simpler, are there are drawbacks that would stop us going down that root?